src/mapgen_v6.cpp

```diff
@@ -836,13 +836,17 @@ void MapgenV6::flowMud(s16 &mudflow_minpos, s16 &mudflow_maxpos)
 					v3s16(-1, 0, 0), // left
 				};
 
-				// Check that upper is air or doesn't exist.
-				// Cancel dropping if upper keeps it in place
+				// Check that upper is walkable. Cancel
+				// dropping if upper keeps it in place.
 				u32 i3 = i;
 				vm->m_area.add_y(em, i3, 1);
-				if (vm->m_area.contains(i3) == true &&
-						ndef->get(vm->m_data[i3]).walkable)
-					continue;
+				MapNode *n3 = NULL;
+
+				if (vm->m_area.contains(i3) == true) {
```

Comment:
It's because the existing code is full of '== false' and '== true'. Maybe after release i will go through mgv6 code and clean it up.